web-animations-1/Overview.bs

```diff
@@ -1150,7 +1125,14 @@ as CSS Animations [[CSS-ANIMATIONS-1]].
     condition from the following, if any:
 
     <div class="switch">
-
+    :   If |animation|'s <a>current time</a> is <a>unresolved</a>,
+        <a>start time</a> is <a>unresolved</a>, the <var>auto-rewind</var>
+        flag is true and <em>either</em>:
+        *   the |animation| has no associated <a>timeline</a>, or
+        *   the associated <a>timeline</a> is
+            <a lt="inactive timeline">inactive</a>
```

Comment:
I'm afraid I don't quite understand the reasoning behind these conditions. I thought we discussed setting the start time to zero for scroll-based timelines? But they won't necessarily have no timeline, right?

Reply:
Setting start time to zero and hold time to null produces unresolved current time. This is to ensure no animation effect is applied when the timeline is inactive.

Reply:
Just to be clear regarding your comment about zero start time initialization for scroll timeline, I am planning to make this change later, but as part of a separate PR. The goal of this PR is just handling inactive/null timelines.

Reply:
I see. So the tricky part is making an inactive timeline produce an unresolved current time in the animation whilst also still allowing animations without a timeline to be seeked to a particular time.

This is complicated because we might have Web compat issues here given that Chrome and Firefox both ship the `Animation()` constructor so the behavior of calling `play()` on animations without a timeline is already observable.

That said, we probably don't commonly encounter inactive timelines, so I wonder if we can make this change simply apply to inactive timelines?

Or is it a bit odd that calling `play()` on an animation without a timeline behaves differently from calling `play()` on an animation attached to an inactive timeline? From my understanding, both would make the animation "running" and pending, but in the first case the `currentTime` would be zero, but in the second case the `currentTime` would be `null`.

Reply:
With #4899 proposal, if accepted, play/pause procedures will be updated to initialize start time to zero for both active and inactive finite timelines. Event without #4899, we probably should be initializing start time to 0 for finite timeline linked animations regardless of the timeline activeness.

Having said this, we need to decide which behavior we expect from animations without a timeline. In #2066 the "half" agreement was to treat a null timeline as inactive timeline, meaning animations don't expose animation effect. If this indeed the desired behavior, perhaps we can find a way to phaseout already shipped behavior in favor of the desired one.

Reply:
Changed to initialize start time of finite timeline animations to 0. No changes are made for the animations that don't have a timeline. Note, in this change we initialize start time to 0 regardless of the timeline activeness. It achieves the desired effect when the timeline is active (anim.currentTime==timeline.currentTime) and when the timeline is inactive (animation.currentTime = null).